### PR TITLE
fixtures for exportService unit tests

### PR DESCRIPTION
**Pull request by TimCsaky**
# Description
ticket: [https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-2129](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-2129)

Here are fixtures for 
- the 'kitchen sink' schema that contains all the 'simple' (basic) fields and layout components
- an 'advanced' form schema that contains all the  remaining advanced fields, data, layout and bcgov components
- a submission from each form (1:1 copy of the form_submission.submission db field)
- 'submissions export' (the json that is downloaded through the ui when user does a submissions export)

I'm using these fixtures to verify the submissions export to CSV is building the columns in the right order and not including unwanted fields. I've changed the recursive _readSchemaFields method to also find some of the more advanced field names. (datagrid, editgrid, selectboxes, etc). This is definitely a best effort., some field id's aren't available in the form schema and are appended to the end of the list of columns.

Getting this list of fields is something we could put in a shared util(?) method and use when adding custom columns to the 'all submissions' table, cdogs template etc. 

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-238

**Review by loneil — APPROVED** [review]
Looks good

File: app/src/forms/form/exportService.js [hunk 1]
```diff
@@ -65,27 +65,59 @@ const service = {
...
-      // Handle current object level
-      if (obj.key && obj.input && !obj.hidden) {
-        // Add current field key if it is non-hidden input field
-        fields.push(obj.key);
-        // Look-ahead and add all sub-values as fields if type is simplecheckbox or checkbox
-        // TODO: there may be more cases where attribute is an object tree
-        if (obj.type && obj.type.includes('checkbox')) {
-          if (obj.values) obj.values.forEach(e => fields.push(`${obj.key}.${e.value}`));
+      // if an input component (not hidden or a button)
+      if (obj.key && obj.input && !obj.hidden && obj.type!== 'button') {
```
Comment (jujaga) [inline app/src/forms/form/exportService.js:71]: Missing space between type and !==

File: app/src/forms/form/exportService.js [hunk 2]
```diff
@@ -65,27 +65,59 @@ const service = {
...
+        // if the fieldname we want is defined in component's sub-values
+        const componentsWithSubValues = ['simplecheckboxes', 'selectboxes', 'survey', 'address'];
+        if (obj.type && componentsWithSubValues.includes(obj.type)) {
+          // for survey component, get field name from obj.questions.value
+          if (obj.type === 'survey') {
+            obj.questions.forEach(e => fields.push(`${obj.key}.${e.value}`));
+          }
+          // for checkboxes and selectboxes, get field name from obj.values.value
+          else if (obj.values) {
+            obj.values.forEach(function (e) {
```
Comment (jujaga) [inline app/src/forms/form/exportService.js:82]: Arrow function

**Review by jujaga — APPROVED** [review]
A few minor things to fix but looks good otherwise. 👍

**Pull request merged**